### DISCUSSION TITLE
Call to a member function routes() on null

### DISCUSSION THREAD

**Discussion by m85eu** [codeigniter4/shield#221] [Q&A]
Hi

Im trying to migrate from myth auth to shield however I've stucked at very beginning.

After install I've got an error:
```
Call to a member function routes() on null 

APPPATH/Config/Routes.php at line 229
service('auth')->routes($routes);
```

**Comment by kenjis** [comment]
How did you install Shield?

**Reply by m85eu** [reply]
Via composer.

**Reply by datamweb** [reply]
Hello @m85eu , 
what are you typing in the terminal?
Which version did you install?
Do you currently have `myth-auth` installed in your CI project?
And what did you do after installation shield?

**Reply by m85eu** [reply]
My installation step by step (CI 4.2.0)
1.
`composer require codeigniter4/shield:dev-develop`
2.
`php spark shield:setup`
3. After that I changed a little bit Config/Auth and Config/AuthGroups:
from `CodeIgniter\Shield\Config\Auth as ShieldAuth;` to `use CodeIgniter\Shield\Config\Auth as ShieldAuth;`
from `CodeIgniter\Shield\Config\AuthGroups as ShieldAuthGroups;` to `use CodeIgniter\Shield\Config\AuthGroups as ShieldAuthGroups;`

Thanks in advance for any help.

![err](https://user-images.githubusercontent.com/64414695/172775172-0cd86405-3c61-41fe-8c6a-1a98ba21be85.png)

**Reply by kenjis** [reply]
> Call to a member function routes() on null 

It means `service('auth')` returns `null`.
It seems your CodeIgniter configuration is different from the default.
Did you customize `Config\Modules`?

**Reply by m85eu** [reply]
No, I didnt change anything in Config\Modules.

**Reply by kenjis** [reply]
What if you install CI4 and Shield from the scratch?
The same error shows?
